src/clients/googleClassroom.js

```diff
@@ -8,3 +13,90 @@ export function createShareToClassroomUrl(snapshotKey, title) {
   uri.search = `snapshot=${snapshotKey}`;
   return BASE_URL + qs.stringify({url: uri.href, title});
 }
+
+export function createSnapshotUrl(assignemtnKey) {
+  const uri = document.createElement('a');
+  uri.href = '/';
+  uri.search = `snapshot=${assignemtnKey}`;
+  return uri.href;
+}
+
+export async function getCourses(accessToken) {
+  const gapi = await initGapi();
+  const client = await initGapiClient(gapi);
+  const classroom = await initGapiClientClassroom(client, accessToken);
+  const courses = await classroom.courses.list();
+  return courses.result.courses;
+}
+
+export async function createClassroomCourseWork(
+  accessToken,
+  workType,
+  courseId,
+  url,
+  title,
+) {
+  const gapi = await initGapi();
+  const client = await initGapiClient(gapi);
+  const classroom = await initGapiClientClassroom(client, accessToken);
+  const newCoursework = await classroom.courses.courseWork.create({
+    courseId,
+    title,
+    workType,
+    state: 'PUBLISHED',
```

Comment:
For the purposes of this PR, let’s publish everything as a draft, and then have a bigger discussion about whether to rebuild the due date/scheduling interface etc.